src/avram/pool_statement_logging.cr

```diff
@@ -0,0 +1,44 @@
+module DB
+  abstract class PoolStatement
+    def exec : ExecResult
+      log_query
+      statement_with_retry &.exec
+    end
+
+    def exec(*args) : ExecResult
+      log_query(args)
+      statement_with_retry &.exec(*args)
```

Comment:
Unfortunately `super` does not work with an abstract class and an abstract module which  is included in the `PoolStatement`. I feel ok copying the methods since we have extensive query tests that should catch any issues, and because these methods are extremely simple.